logic/LogicCache.java

```diff
@@ -31,15 +33,18 @@ public class LogicCache {
 
     private final CommonCache<GraphTraversal.Type, Optional<Map<Identifier.Variable.Retrievable, Set<Label>>>> typeInferenceCache;
     private final CommonCache<String, Rule> ruleCache;
+    private final CommonCache<Concludable, Map<Rule, Set<Unifier>>> applicableRules;
```

Comment:
Add the cache to use for all queries on which rules a concludable triggers and with what unifiers.